web/tests/lib/example_user.js

```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+let last_issued_user_id = 1000;
+
+const get_user_id = () => {
+    last_issued_user_id += 1 + Math.floor(Math.random() * 100);
+    return last_issued_user_id;
+};
+
+const bot_or_user_props = (opts = {}) => {
+    // Since other fields need `user_id`, we extract it early.
+    const user_id = opts.user_id ?? get_user_id();
+
+    const common_props = {
+        user_id,
+        delivery_email: opts.delivery_email ?? null,
+        email: `user-${user_id}@example.org`,
+        full_name: `user-${user_id}-ex_name`,
+        date_joined: Date.now(),
+        is_owner: false,
+        is_admin: false,
+        is_guest: false,
+        timezone: "UTC",
+        avatar_version: 0,
+        // By default a member.
+        role: 400,
```

Comment:
Should we make values like `role` and `bot_type` enums? Since it's not really clear what number represents which role.

Reply:
I think we won't be able to get the most out of this change, since we are defaulting the field to any one value/branch of the enum all the time, hence all other (~3-5) branches would be of no use -- nevertheless this is a clear improvement in terms of readability, so I am making the change :)